### ISSUE TITLE
fix crash when processing incomplete ClientHello 

### ISSUE THREAD

**Issue by astibal** [astibal/smithproxy#6]
reproduced with:
`
sslyze --regular --starttls=<server>:587
`
got crash on 
``` * ROBOT Attack:
Unhandled exception while running --robot:  
ConnectionResetError - [Errno 104] Connection reset by peer
```
smithproxy crashed and restarted

**Comment by astibal** [comment]
fix 602c33a8e6eb89133321c2bd40b548d9cb11872e confirmed for transparent proxy

**Comment by astibal** [comment]
old and fixed
